### PR TITLE
New release of Firebird Driver see #710

### PR DESCRIPTION
**Pull request by mnewnham**
This new release of the driver for Firebird is designed for use with the latest PHP7 drivers.

### CODE REVIEW

File: drivers/adodb-firebird.inc.php [hunk 1]
```diff
@@ -255,37 +355,65 @@ function &MetaIndexes ($table, $primary = FALSE, $owner=false)
...
-		$seqname = strtoupper($seqname);
-		return $this->Execute("DROP GENERATOR $seqname");
-	}
-
-	function GenID($seqname='adodbseq',$startID=1)
+		$sql = sprintf($this->_genSeqSQL,$seqname,$startID);
+		$ok = $this->Execute($sql);
+		
+		if (!$ok) 
+			return false;
```
Comment (dregad) [inline drivers/adodb-firebird.inc.php:398]: Aren't we missing a `return $ok;` statement here ? The PHPDoc says the function returns a recordset.

File: drivers/adodb-firebird.inc.php [hunk 2]
```diff
@@ -255,37 +355,65 @@ function &MetaIndexes ($table, $primary = FALSE, $owner=false)
...
+	
+	/**
+	 * A portable method of creating sequence numbers.
+	 *
+	 * @link https://adodb.org/dokuwiki/doku.php?id=v5:reference:connection:genid
+	 *
+	 * @param string $seqname (Optional) The name of the sequence to use.
+	 * @param int $startID (Optional) The point to start at in the sequence.
+	 *
+	 * @return bool|int|string
```
Comment (dregad) [inline drivers/adodb-firebird.inc.php:409]: Looks like we're only returning _int_ here

File: drivers/adodb-firebird.inc.php [hunk 3]
```diff
@@ -307,23 +435,36 @@ function SelectDB($dbName)
...
+	/**
+	 * Prepares an SQL statement and returns a handle to use.
+	 * This is not used by bound parameters anymore
+	 *
+	 * @link https://adodb.org/dokuwiki/doku.php?id=v5:reference:connection:prepare
+	 * @todo update this function to handle prepared statements correctly
+	 *
+	 * @param string $sql The SQL to prepare.
+	 *
+	 * @return string The original SQL that was provided.
```
Comment (dregad) [inline drivers/adodb-firebird.inc.php:465]: The function actually returns an array

File: drivers/adodb-firebird.inc.php [hunk 4]
```diff
@@ -610,105 +816,106 @@ function UpdateBlobFile($table,$column,$path,$where,$blobtype='BLOB')
...
+	* Returns a portably-formatted date string from a timestamp database column.
+	*
+	* @link https://adodb.org/dokuwiki/doku.php?id=v5:reference:connection:sqldate
+	*
+	* Firebird does not support an AM/PM format, so the A indicator always shows AM
+	*
+	* @param string $fmt The date format to use.
+	* @param string|bool $col (Optional) The table column to date format, or if false, use NOW().
+	*
+	* @return bool|string The SQL DATE_FORMAT() string, or false if the provided date format was empty.
```
Comment (dregad) [inline drivers/adodb-firebird.inc.php:874]: Unless I missed something, the function never returns _false_

**Review by dregad — COMMENTED** [review]
I noted a few minor issues / inconsistencies in PHPDoc

**Comment by mnewnham** [comment]
Ok, thanks for the review. It may be that some of these existed already and I didn't notice. I'll check them later

File: drivers/adodb-firebird.inc.php (hunk 1 above)
Comment (mnewnham) [reply drivers/adodb-firebird.inc.php]: should return a recordset

File: drivers/adodb-firebird.inc.php (hunk 2 above)
Comment (mnewnham) [reply drivers/adodb-firebird.inc.php:406]: Updated doc

File: drivers/adodb-firebird.inc.php (hunk 3 above)
Comment (mnewnham) [reply drivers/adodb-firebird.inc.php:462]: Returns an array, or false if the preparation fails

File: drivers/adodb-firebird.inc.php (hunk 4 above)
Comment (mnewnham) [reply drivers/adodb-firebird.inc.php:871]: The docblock for this is misleading for this method is misleading in both the prototype and a number of the drivers

**Pull request merged**